lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -268,8 +268,11 @@ def get(self, request, *args, **kwargs):  # pylint: disable=too-many-statements
                     'Please contact your degree administrator or '
                     '{platform_name} Support if you have questions.'
                 ).format(platform_name=settings.PLATFORM_NAME)
-            elif course_is_invitation_only(course):
+            elif CourseEnrollment.is_enrollment_closed(request.user, course_overview):
```

Comment:
What this new code does is:
- `CourseEnrollment.is_enrollment_closed` →
- `user.has_perm(ENROLL_IN_COURSE, course)` →
- `has_access(user, 'can_enroll', course)`  →
- `_can_enroll_courselike(user, course)`

And that method checks:
- if the user is allow-listed via a `CourseEnrollmentAllowed` entry (NEW)
- if the user has staff access
- if the course is a ccx course (NEW and added by me here in this PR)
- if the course is invitation only
- if we are outside of the enrollment window (NEW)